app/Services/AppointmentService.php

```diff
@@ -118,9 +118,20 @@ public function reject($data, $appointmentId)
 
     public function checkAvailabiltyToReject($appointment)
     {
-        $nowInMs = Carbon::now()->timestamp * 1000;
+        $nowInMs = Carbon::now()->timestamp;
         if ($appointment->status != AppointmentStatus::PENDING || $appointment->time <= $nowInMs) {
             abort(Response::HTTP_BAD_REQUEST,"already rejected or expired");            
         }
     }
+
+    public function complete($appointment)
+    {
+        $nowInMs = Carbon::now()->timestamp;
+        $appointmentFinishedAt = ($appointment->time + ($appointment->duration * 60));
+        if ($appointmentFinishedAt > $nowInMs || $appointment->status != AppointmentStatus::CHECKED || $appointment->status == AppointmentStatus::COMPLETED) {
```

Comment:
No need to check the status complete as you already checked `$appointment->status != AppointmentStatus::CHECKED`